Tests/CSharp/TypeCastTests.cs

```diff
@@ -296,18 +717,38 @@ internal enum TestEnum2
 
 internal partial class Class1
 {
-    private void Test()
+    private void TestIntegrals(byte b, short s, int i, long l, TestEnum2 e)
     {
-        TestEnum res = (TestEnum)5;
-        res = (TestEnum)Conversions.ToInteger(5m);
-        res = (TestEnum)5L;
-        res = (TestEnum)Conversions.ToInteger(5.4f);
-        res = (TestEnum)Conversions.ToInteger(5.7d);
-        res = (TestEnum)5UL;
-        res = (TestEnum)5;
+        TestEnum? res = (TestEnum?)b;
+        res = (TestEnum?)s;
+        res = (TestEnum?)i;
+        res = (TestEnum?)l;
+        res = (TestEnum?)e;
+    }
+
+    private void TestNullableIntegrals(byte? b, short? s, int? i, long? l, TestEnum2? e)
+    {
+        TestEnum? res = (TestEnum?)b;
```

Comment:
This one is actually not true but I will fix that in another PR as this one is already big enough.
```csharp
TestEnum res = (TestEnum)b.Value;
```